local_sales/app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CategoriesController < ApplicationController
+  before_action :set_product, only: %i[show edit update destroy]
```

Comment:
this won't work, the method must be *set_product* as you defined at your private methods